File: Easy/1437. Check If All 1's Are at Least Length K Places Away.java
class Solution {
    public boolean kLengthApart(int[] nums, int k) {
        int last = -k - 2;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == 0) {
                continue;
            }
            if (i - last <= k) {
                return false;
            }
            last = i;
        }
        return true;
    }
}

File: Easy/1437. Check If All 1's Are at Least Length K Places Away.py
from typing import List


class Solution:
    def kLengthApart(self, nums: List[int], k: int) -> bool:
        if k == 0:
            return True
        last_idx = -k - 1
        for i, v in enumerate(nums):
            if v == 1:
                if i - last_idx <= k:
                    return False
                last_idx = i
        return True